docs/proposed/pub-sub.rst

```diff
@@ -0,0 +1,124 @@
+.. -*- coding: utf-8 -*-
+
+Pub/Sub for Tahoe-LAFS
+======================
+
+There are various use-cases in Tahoe-LAFS where it would be
+advantagous to send messages to particular clients (or for particular
+topics). A well-known pattern called "Publish/Subscribe" or "PubSub"
+would answer many of these use-cases, but Tahoe-LAFS does not include
+such messaging.
+
+
+Use Cases
+---------
+
+Many of these use-cases are special-cases of a general want:
+
+ - there exists a mutable capability;
+ - one client updates content in this capability;
+ - ..and one or more (different) clients wishes to know when it changes
+
+Thus, a "pub/sub" pattern could be implemented by adding a mechanism
+to notify clients that a particular mutable (to which they possess a
+valid read-capability) has changed. This would maintain many of the
+existing properties of capabilities.
+
+
+Magic Folders Updates
+`````````````````````
+
+Magic Folders in Tahoe currently poll, looking for updates. This
+involves periodically downloading some mutable directory-capabilities
+and determining if anything has changed. This is wasteful of bandwidth
+and server resources. It would nice if a client, Alice, could tell
```

Comment:
It is wasteful of _client_ resources too. :)

(Maybe remove "server" and just say "resources"?)